### ISSUE TITLE
App/Window Switching Similar to MacOS

### ISSUE THREAD

**Issue by GRMrGecko** [pop-os/cosmic-launcher#218]
Hello,

I am opening this issue here as research shows a request for this feature https://github.com/pop-os/cosmic-epoch/issues/490 which was told for organization reasons to open a request under cosmic-launcher. I tried searching it see if someone already opened an issue here regarding it, but did not see anything.

I'd like to see the option of adding:

1. A short cut to just switch between windows of the same application. In MacOS, this is ``Alt+` `` and for reverse order is `Alt+~`.
2. The ability to limit the default task switcher to one window per application. MacOS does this, and I for one prefer this option as to having many windows opened in one app floods the task switching.

Other platforms that offer these options include KDE Plasma, and I believe Gnome.

**Comment by pheki** [comment]
> Other platforms that offer these options include KDE Plasma, and I believe Gnome.

Yes, GNOME offers this option and it's the default behavior for both Alt+TAB and Super+TAB.

Default dconf values for GNOME:
- `/org/gnome/desktop/wm/keybindings/switch-applications`: `['<Super>Tab', '<Alt>Tab']`
- `/org/gnome/desktop/wm/keybindings/switch-windows`: `[]`

**Comment by iamkartiknayak** [comment]
`Super + Tab` for switching distinct apps
`Alt + Tab` for switching different instance of same app

using `<Key> + Shift + tab` to reverse the operation for both.

**Comment by pheki** [comment]
> `Super + Tab` for switching distinct apps 
> `Alt + Tab` for switching different instance of same app
> 
> using `<Key> + Shift + tab` to reverse the operation for both.

I personally feel like using `Tab`  and `Above_Tab` (\` in the US mapping) is slightly better because you can navigate all at once. 

e.g. let's say you want to go to a specific firefox window

1. Hold Alt
2. Press Tab until you reach Firefox
3. Press Above_Tab (\`) until you reach the window you want
4. Release Alt

This is how it works on Gnome and MacOS.

That said, `Super + Tab` and `Alt + Tab` should work fine, especially if it's configurable.
